CompanyEmployees.Application/Validators/CheckInAttendanceCommandValidator.cs

```diff
@@ -0,0 +1,37 @@
+using CompanyEmployees.Application.Commands;
+using FluentValidation;
+using FluentValidation.Results;
+
+namespace CompanyEmployees.Application.Validators;
+
+public sealed class CheckInAttendanceCommandValidator : AbstractValidator<CheckInAttendanceCommand>
+{
+    public CheckInAttendanceCommandValidator()
+    {
+        RuleFor(a => a.Attendance.EmployeeId)
+            .NotEmpty()
+            .WithMessage("Employee ID is required.");
+
+        RuleFor(a => a.Attendance.CheckInTime)
+            .NotEmpty()
+            .WithMessage("Check-in time is required.");
+
+        RuleFor(a => a.Attendance.Status)
+            .NotEmpty()
+            .MaximumLength(10)
+            .WithMessage("Status is required and must not exceed 10 characters.");
+
+        RuleFor(a => a.Attendance.Notes)
+            .MaximumLength(200)
+            .When(a => !string.IsNullOrEmpty(a.Attendance.Notes))
+            .WithMessage("Notes must not exceed 200 characters.");
+    }
+
+    public override ValidationResult Validate(ValidationContext<CheckInAttendanceCommand> context)
```

Comment:
Consider moving the null check for `Attendance` into a `RuleFor(a => a.Attendance)` with a `NotNull()` rule. This avoids having to override the `Validate` method and improves readability.

```suggestion
RuleFor(a => a.Attendance)
    .NotNull().WithMessage("AttendanceForCreationDto object is null");
```